Tests/Integration/AuthorizationServerTest.php

```diff
@@ -103,6 +103,21 @@ public function testRestrictedGrantClient(): void
         $this->assertSame('Client authentication failed', $response['message']);
     }
 
+    public function testRestrictedScopeClient(): void
+    {
+        $request = $this->createAuthorizationRequest('quux_restricted_scopes:beer', [
+            'grant_type' => 'client_credentials',
+            'scope' => 'fancy rock',
```

Comment:
@spideyfusion Can you please use fixture constants instead of concrete values? Now it is mixed between two and makes it hard to track.

Reply:
@alenpokos This will be resolved in a separate PR where we'll tackle test readability.